### PR TITLE
Less psalm warnings

### PR DESCRIPTION
**Pull request by kesselb**
- Update signature for like/notLike/in/notIn: **$x** is always passed to quoteColumnName hence ILiteral|IParameter|IQueryFunction are supported.
- Update signature for in/notIn: **$y** is always passed to quoteColumnName hence ILiteral|IParameter|IQueryFunction are supported.
- Update signature for set: also ILiteral and IQueryFunction are valid values.
- Update signature for setValue: also IParameter is supported.

:warning: Functional change: setValue now casts $value to string. This was done before much later by doctrine.

### CODE REVIEW

File: lib/private/DB/QueryBuilder/QueryBuilder.php [hunk 1]
```diff
@@ -868,14 +869,14 @@ public function addGroupBy(...$groupBys) {
 	 * </code>
 	 *
 	 * @param string $column The column into which the value should be inserted.
-	 * @param string $value The value that should be inserted into the column.
+	 * @param IParameter|string $value The value that should be inserted into the column.
```
Comment (MorrisJobke) [inline lib/private/DB/QueryBuilder/QueryBuilder.php:872]: Shouldn't then the upstream PHPDoc be updated? Because then it wouldn't need this hack here.

Comment (kesselb) [reply lib/private/DB/QueryBuilder/QueryBuilder.php:872]: Upstream? The doctrine world does not know IParameter.

Comment (MorrisJobke) [reply lib/private/DB/QueryBuilder/QueryBuilder.php:872]: 🙈 Ah ... just noticed that this is our interface :/ Okay then this is fine.

**Review by MorrisJobke — APPROVED** [review]
So far looks good 👍 

Also neat that it solves 129 errors :)

File: lib/private/DB/QueryBuilder/ExpressionBuilder/ExpressionBuilder.php [hunk 2]
```diff
@@ -278,7 +279,7 @@ public function isNotNull($x) {
 	/**
 	 * Creates a LIKE() comparison expression with the given arguments.
 	 *
-	 * @param string $x Field in string format to be inspected by LIKE() comparison.
+	 * @param ILiteral|IParameter|IQueryFunction|string $x Field in string format to be inspected by LIKE() comparison.
```
Comment (nickvergessen) [inline lib/private/DB/QueryBuilder/ExpressionBuilder/ExpressionBuilder.php:282]: I think this needs to be done on some other functions too, but well one after the other

**Comment by MorrisJobke** [comment]
@kesselb the PHPDoc of setValue is updated, right? Anything else missing to get this in?

**Review by nickvergessen — APPROVED** [review]

**Comment by kesselb** [comment]
> @kesselb the PHPDoc of setValue is updated, right? Anything else missing to get this in?

@MorrisJobke ready. Just the check was missing.

**Comment by ChristophWurst** [comment]
:pray: this also solves some issues in Mail reported by psalm. Thanks :rocket:

**Comment by MorrisJobke** [comment]
I back ported the PHPDoc in #23971, because there was the method newly used and thus caused issues.

**Pull request merged**